lucille-core/src/main/java/com/kmwllc/lucille/stage/DropIfFieldEquals.java

```diff
@@ -0,0 +1,55 @@
+package com.kmwllc.lucille.stage;
+
+import com.kmwllc.lucille.core.ConfigUtils;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.typesafe.config.Config;
+
+import java.util.Iterator;
+
+/**
+ * This stage will drop a document if it contains a given field and the value of that field matches a given value.
+ * Stage gives an option to do a case-sensitive comparison but is not case-sensitive by default.
+ * Stage also gives an option to invert the match, so that the document is dropped if the value does not match.
+ */
+public class DropIfFieldEquals extends Stage {
```

Comment:
Stages already support conditional execution. See the examples in the StageTest resources directory. I have included one example below.

I believe you can just make this stage drop every document it sees... but rely on conditions to prevent it from acting on any document that shouldn't be dropped.

For case-insensitivity, might need to use this together with NormalizeText to lowercase the field value... or investigate whether there's a way to add-case insensitivity as an option in the conditions framework.

conditions = [
  {
    fields = ["user_id", "state", "country"]
    values = ["MA", "1234", "US", "Russia", "4567"]
    operator = "must_not"
  }
]